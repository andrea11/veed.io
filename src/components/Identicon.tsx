import {identicon} from "minidenticons";

type IdenticonProps = {
    username: string;
    saturation?: number;
    lightness?: number;
}

const defaultSaturation = 90;
const defaultLightness = 50;

export default function Identicon(props: IdenticonProps) {
    return (
        <div
            dangerouslySetInnerHTML={{
                __html: identicon(
                    props.username,
                    props.saturation ?? defaultSaturation,
                    props.lightness ?? defaultLightness
                )
            }}
        />
    );
}