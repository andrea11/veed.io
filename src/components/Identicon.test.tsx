import Identicon from "@/components/Identicon";
import {render} from "@testing-library/react";

jest.mock("minidenticons", () => ({
    identicon: jest.fn()
}))

describe("Identicon", () => {
    it("renders correctly", () => {
        const { container } = render(<Identicon username={"username"}/>)
        expect(container).toMatchSnapshot()
    })
})