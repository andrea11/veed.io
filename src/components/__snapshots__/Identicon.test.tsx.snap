// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Identicon renders correctly 1`] = `
<div>
  <div />
</div>
`;
